src_asm/reading_binary_utils.c

```diff
@@ -0,0 +1,56 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   reading_binary_utils.c                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: dtimeon <dtimeon@student.42.fr>            +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/05/16 01:21:52 by dtimeon           #+#    #+#             */
+/*   Updated: 2020/05/17 23:00:50 by dtimeon          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+static ssize_t		get_champ_code_size(t_file *file, ssize_t header_size)
+{
+	ssize_t			code_size;
+
+	code_size = lseek(file->fd, 0, SEEK_END) - header_size;
+	if (code_size >= 0 &&
+		(lseek(file->fd, header_size, SEEK_SET)) >= 0)
+		return (code_size);
+	else
+		return (-1);
+}
+
+static void			save_binary_champ_code(t_file *file, ssize_t code_size)
+{
+	unsigned char	*buffer;
+	
+	buffer = (unsigned char *)malloc(sizeof(unsigned char) * code_size);
+	if (!buffer)
+		exit_with_allocation_error(file->filename);
+	if (read(file->fd, buffer, code_size) >= 0)
+	{
+		file->champ_code = buffer;
+		file->code_size = code_size;
+		file->is_read_successfully = 1;
```

Comment:
For more readability we can use constants FALSE 0, TRUE 1.

Reply:
Thank you. Done!